src/main/webapp/app/shared/modal/AddMutationModal.tsx

```diff
@@ -80,507 +82,747 @@ function AddMutationModalContent({
+  const fetchNormalAlterationDebounced = useCallback(
+    _.debounce((newAlteration: string, tabIndex: number, alterationData: AlterationData[]) => {
+      fetchNormalAlteration(newAlteration, tabIndex, alterationData);
+    }, 1000),
+    [tabStates.length]
+  );
 
-    const newAlterations = alterations.map((alteration, index) => {
-      return {
-        label: mutationOptionsToAdd[index].label,
-        value: mutationOptionsToAdd[index].label,
-        uuid: mutationOptionsToAdd[index].uuid,
-        Alteration: mutationOptionsToAdd[index].label,
-        Name: alteration.entity?.name || '',
-        Protein_Change: alteration.entity?.proteinChange,
-        Protein_Start: alteration.entity?.start?.toString() || '',
-        Protein_End: alteration.entity?.end?.toString() || '',
-        Ref_Residues: alteration.entity?.refResidues || '',
-        Var_Residues: alteration.entity?.variantResidues || '',
-        Consequence: alteration.entity?.consequence?.name || '',
-        Comment: comment,
-        Excluding: excludingAlterations.map(excludingAlteration => ({
-          label: excludingAlteration.entity?.alteration,
-          value: excludingAlteration.entity?.alteration,
-          uuid: generateUuid(),
-          Alteration: excludingAlteration.entity?.alteration || '',
-          Name: excludingAlteration.entity?.name || '',
-          Protein_Change: excludingAlteration.entity?.proteinChange,
-          Protein_Start: excludingAlteration.entity?.start?.toString() || '',
-          Protein_End: excludingAlteration.entity?.end?.toString() || '',
-          Ref_Residues: excludingAlteration.entity?.refResidues || '',
-          Var_Residues: excludingAlteration.entity?.variantResidues || '',
-          Consequence: excludingAlteration.entity?.consequence?.name || '',
-          Excluding: [],
-          warning: excludingAlteration.warning ? excludingAlteration.message : null,
-          error: excludingAlteration.error ? excludingAlteration.message : null,
-        })),
-        warning: alteration.warning ? alteration.message : null,
-        error: alteration.error ? alteration.message : null,
-      };
+  function handleNormalAlterationChange(newValue: string, tabIndex: number) {
+    setTabStates(states => {
+      const newStates = _.cloneDeep(states);
+      newStates[tabIndex].alterationFieldValueWhileFetching = newValue;
+      return newStates;
     });
+    fetchNormalAlterationDebounced(newValue, tabIndex, tabStates);
+  }
 
-    function replaceOldMutations(oldMutationStates: MutationState[], newMutations: MutationState[], uuidToReplace: string) {
-      return oldMutationStates.reduce((accumulator: MutationState[], currentMutation) => {
-        if (uuidToReplace === currentMutation.uuid) {
-          return [...accumulator, ...newMutations];
-        }
-        return [...accumulator, currentMutation];
-      }, []);
+  async function fetchExcludedAlteration(
+    newAlteration: string,
+    tabIndex: number,
```

Comment:
I'm a bit confused through out this file because of this naming. The tab concept is also used in excluding. Should we rename to alterationIndex?

Reply:
Yeah will definitely do that. I see how it's confusing,